File: 0692-top-k-frequent-words/0692-top-k-frequent-words.java
class Solution {
    public List<String> topKFrequent(String[] words, int k) {
        Map<String,Integer> map = new HashMap<>();
        for(String s:words){
            int a = map.getOrDefault(s,0)+1;
            map.put(s,a);
        }
        List<pair> l = new ArrayList<>();
        for(String st:map.keySet()){
            l.add(new pair(st,map.get(st)));
        }
        Collections.sort(l,(a,b)->a.v==b.v?a.s.compareTo(b.s):b.v-a.v);
        List<String> list = new  ArrayList<>();
        for(int i=0;i< k;i++) list.add(l.get(i).s);
        return list;
    }
    class pair{
        String s;
        int v;
        pair(String s, int v){
            this.s = s;this.v = v;
        }
    }
}

File: 0692-top-k-frequent-words/0692-top-k-frequent-words.cpp
class Solution {
public:
    static bool cmp(pair<int, string> p1, pair<int, string> p2) {
        if(p1.first == p2.first) {
            return p1.second < p2.second;
        }
        else {
            return p1.first > p2.first;
        }
    }
    vector<string> topKFrequent(vector<string>& words, int k) {
        map<string, int> mp;
        for(auto wd : words) {
            mp[wd]++;
        }
        vector<pair<int, string>> v;
        for(auto elem : mp) {
            v.push_back({elem.second, elem.first});
        }
        sort(v.begin(), v.end(), cmp);
        vector<string> ans;
        for(int i = 0; i < k; ++i) {
            ans.push_back(v[i].second);
        }
        return ans;
    }
};